### PR TITLE
opt in and out handled with middleware stack

### PR DESCRIPTION
**Pull request by wheresrhys**
@matthew-andrews @aintgoin2goa @commuterjoy 

Big changes here, but passes all tests aside from those related to tracking (which I will address next)

It's platinum so please feedback in spades

### CODE REVIEW

**Comment by wheresrhys** [comment]
Also, this recreates existing behaviour. However it was so convoluted I think it's likely that bits of functionality that should've been implemented for each route weren't. Now using middleware it's easy to rectify this so please leave comments where you think middleware should be added on a route

**Comment by wheresrhys** [comment]
Also removed all integration with session service as it doesn't appear to be important. By passing cookies & headers to spoor we can get all the tracking we need

File: Makefile [hunk 1]
```diff
@@ -11,14 +11,14 @@ clean:
 install:
 	obt install --verbose
 
-unit-test:
+test-unit:
```
Comment (matthew-andrews) [inline Makefile]: o.O?

Comment (wheresrhys) [inline Makefile]: O.o!

Comment (matthew-andrews) [inline Makefile]: why the rename?

Comment (wheresrhys) [inline Makefile]: In some other apps there are multiple tasks like test-unit, test-e2e, test-client, test etc... It's marginally less typing to change from running one test task to another if the variant is at the end of the task

Happy to change back if there's a good reason to

Comment (matthew-andrews) [inline Makefile]: no it's fine, just curious

Comment (matthew-andrews) [inline Makefile]: we should roll this out more widely, i'll update the ones i see when i see them

**Pull request merged**